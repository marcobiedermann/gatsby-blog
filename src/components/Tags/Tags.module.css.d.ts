declare const styles: {
  readonly tags: string;
  readonly tag: string;
};
export = styles;
