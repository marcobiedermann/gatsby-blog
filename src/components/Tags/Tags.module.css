.tags {
  display: flex;
  gap: 0.5em;
  list-style: none;
  margin-block: 0.5em 1.5em;
  margin-inline: 0;
  padding: 0;
}

.tag {
  background-color: var(--color-headlines);
  border-radius: 999em;
  color: var(--color-background);
  display: inline-block;
  font-size: 0.875rem;
  padding-block: 0.25em;
  padding-inline: 1em;
}
