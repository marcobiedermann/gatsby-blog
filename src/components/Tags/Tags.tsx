import { Link } from 'gatsby';
import React from 'react';
import { tags as tagsStyles, tag as tagStyles } from './Tags.module.css';

export interface TagsProps {
  tags: string[];
}

function Tags(props: TagsProps) {
  const { tags } = props;

  return (
    <ul className={tagsStyles}>
      {tags.map((tag) => (
        <li key={tag}>
          <Link className={tagStyles} to={`/tags/${tag}`}>
            {tag}
          </Link>
        </li>
      ))}
    </ul>
  );
}

export default Tags;
