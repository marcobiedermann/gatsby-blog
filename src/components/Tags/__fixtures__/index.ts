import { TagsProps } from '..';

const tags: TagsProps = {
  tags: ['tag-1'],
};

export default tags;
