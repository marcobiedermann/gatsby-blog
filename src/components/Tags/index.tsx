import { Link } from 'gatsby';
import React, { FC } from 'react';
import * as styles from './style.module.css';

export interface TagsProps {
  tags: string[];
}

const Tags: FC<TagsProps> = (props) => {
  const { tags } = props;

  return (
    <ul className={styles.tags}>
      {tags.map((tag) => (
        <li key={tag}>
          <Link className={styles.tag} to={`/tags/${tag}`}>
            {tag}
          </Link>
        </li>
      ))}
    </ul>
  );
};

export default Tags;
