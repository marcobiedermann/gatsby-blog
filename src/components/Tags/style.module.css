.tags {
  display: flex;
  gap: 0.5em;
  list-style: none;
  margin-bottom: 1.5em;
  margin-left: 0;
  margin-right: 0;
  margin-top: 0.5em;
  padding: 0;
}

.tag {
  background-color: var(--color-headlines);
  border-radius: 999em;
  color: var(--color-background);
  display: inline-block;
  font-size: 0.875rem;
  padding: 0.25em 1em;
}