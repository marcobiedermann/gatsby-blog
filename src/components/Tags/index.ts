import Tags from './Tags';

export * from './Tags';

export default Tags;
