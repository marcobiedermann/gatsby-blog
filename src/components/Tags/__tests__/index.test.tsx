import { render } from '@testing-library/react';
import React from 'react';
import Tags from '..';

describe('tags component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Tags tags={[]} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
