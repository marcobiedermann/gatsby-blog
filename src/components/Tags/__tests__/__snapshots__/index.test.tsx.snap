// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tags component should render correctly 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <a
        href="/tags/tag-1"
      >
        tag-1
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
